liquibase-standard/src/main/java/liquibase/changelog/DatabaseChangeLog.java

```diff
@@ -899,7 +903,12 @@ public SortedSet<Resource> findResources(
             List<Resource> unsortedResources = getUnsortedResources(pathName, errorIfMissingOrEmpty, resourceAccessor, relativeTo, seenChangelogPaths, searchOptions);
             SortedSet<Resource> resources = new TreeSet<>((o1, o2) -> resourceComparator.compare(o1.getPath(), o2.getPath()));
             if (unsortedResources != null) {
+                URI liquibaseHomeInternalUri = getLiquibaseHomeInternalUri();
                 for (Resource resourcePath : unsortedResources) {
+                    // if the resource is inside a jar located in liquibaseHomeUri/internal, we don't want to include it
+                    if (liquibaseHomeInternalUri != null && resourcePath.getUri().toString().startsWith("jar:" + liquibaseHomeInternalUri.toString())) {
+                        continue;
```

Comment:
Can you add logging here? Otherwise this could be a very frustrating edge case for someone to experience.

Reply:
Sure, done!